File: best time to buy and sell stock/sol.java
class Solution {
    public int maxProfit(int[] prices) {
        int size = prices.length;
        int buy = prices[0];
        int max = 0;
        for(int i=1;i<size;i++)
        {
            if(prices[i]<buy)
            {
                buy = prices[i];
            }
            else
            {
                int temp = prices[i] - buy;
                if(temp > max)
                {
                    max = temp;
                }
            }
        }
        return max;
    }
}

File: best time to buy and sell stock/sol.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let len = prices.length;
    let max = 0;
    let buy = prices[0];
    for(let i=1;i<len;i++)
    {
        if(prices[i] < buy)
        {
            buy = prices[i];
        }
        else
        {
            let temp = prices[i] - buy;
            if(temp > max)
            {
                max = temp;
            }
        }
    }
    return max;
};
